### PR TITLE
Add normalized Levenshtein and Damerau-Levenstein

### LINKED ISSUE #19: Normalized Levenshtein score
Hello!

While Levenshtein calculates the minimum number of a string "mutations" there are situations where it's useful to get normalized score, where returned value is calculated as the following:

```rust
1.0 - levenshtein(a, b) / max(a, b);
```

What do you think, is it worth adding such a function here? I could add that. At the same time, it could be a practice for me in writing Rust code :blush:

### PR DESCRIPTION
**Pull request by gentoid**
Fixes #19 

I'm not sure if everything I've added here is necessary. I've done everything just in the likeness. And also added more tests for the corner cases

### CODE REVIEW

File: tests/lib.rs [hunk 1]
```diff
@@ -16,6 +16,31 @@ fn levenshtein_works() {
     assert_eq!(3, levenshtein("kitten", "sitting"));
 }
 
+#[test]
+fn normalized_levenshtein_works() {
+    assert!((normalized_levenshtein("kitten", "sitting") - 0.57142).abs() < 0.00001);
+}
+
+#[test]
```
Comment (dguo) [inline tests/lib.rs]: This is another nitpick, but would you mind only leaving one test for each function in this file? These tests are integration tests that only need to verify that the functions can be called from the outside. You can move the rest of the tests to the bottom of `src/lib.rs` to serve as unit tests.

File: src/lib.rs [hunk 2]
```diff
@@ -171,6 +171,24 @@ pub fn levenshtein(a: &str, b: &str) -> usize {
...
+/// use strsim::normalized_levenshtein;
+///
+/// assert!((normalized_levenshtein("kitten", "sitting") - 0.57142).abs() < 0.00001);
+/// assert!((normalized_levenshtein("", "") - 1.0).abs() < 0.00001);
+/// assert!(normalized_levenshtein("", "second").abs() < 0.00001);
+/// assert!(normalized_levenshtein("first", "").abs() < 0.00001);
+/// assert!((normalized_levenshtein("string", "string") - 1.0).abs() < 0.00001);
+/// ```
+pub fn normalized_levenshtein(a: &str, b: &str) -> f64 {
+    if a.len() == 0 && b.len() == 0 {
```
[the remark this answers is not in the document]
Comment (gentoid) [reply src/lib.rs]: yeah, you're right. `bytes` != `chars` in general case

btw, I just found this method [`&str::is_empty`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.is_empty). For me it looks like even better way here. Yes, it returns `true` for zero **bytes**, but can any string containing multi-byte chars be an empty string at same time?

One benefit I can see here, we don't measure `length` of string, but ask "is this string empty?" Which has more clear semantics

What do you think?

File: tests/lib.rs (hunk 1 above)
Comment (gentoid) [reply tests/lib.rs]: Makes sense. Will do as you suggested :+1:

Comment (gentoid) [reply tests/lib.rs]: Moved  tests in 8f4c33759af0ea84e77f467df306dea5ef727f11

**Comment by gentoid** [comment]
I just realized that I also need to update readme

File: src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dguo) [reply src/lib.rs]: 1. Good idea. I agree that using `is_empty` would have better semantics for this line. Go for it. We'll still have to count the chars for the return expression though.

2. Yeah, I've seen that bit about graphemes before, but I'd rather keep it simple for now. We'd have to use an external crate for it, and I wouldn't be surprised if it's slower than character iteration. Maybe in the future, we can add a way for users to specify the iteration method.

[the remark this answers is not in the document]
Comment (gentoid) [reply src/lib.rs]: Fixes:
- check empty strings by` is_empty` in 13b89d3ecd931b9e131d3a28c7727f0fd19ef2f8
- count chars instead of bytes in 601d4cc86517da781d629fa74a0a63407af19015

**Comment by gentoid** [comment]
@dguo could you please review once more?

**Comment by dguo** [comment]
Thanks for this. The changes are published in version 0.8.0.
